### PR TITLE
feat: modal and navigation components added

### PR DESCRIPTION
**Pull request by hrishi82**
Components added:

- Modal with JS functionality
- Navigation Bar

### CODE REVIEW

File: components/Modal/modal.css [hunk 1]
```diff
@@ -0,0 +1,48 @@
+.modal{
+    display: none;
+    position: fixed;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+    background-color: var(--primary-clr);
```
Comment (PeeyushPrashant) [inline components/Modal/modal.css:7]: Just try using color instead of clr as it will provide more understanding

**Review by PeeyushPrashant — APPROVED** [review]
looks good

**Review by Rohanmond — APPROVED** [review]
Looks good

File: components/Modal/modal.css (hunk 1 above)
Comment (hrishi82) [reply components/Modal/modal.css:7]: noted!

**Pull request merged**